### PR TITLE
perf: Implement only readServiceStr()

### PR DESCRIPTION
**Pull request by Raynos**
This is a subset of https://github.com/uber/tchannel-node/pull/271

This only implements `readServiceStr()`; I've added a benchmark
for `readServiceStr()`.

It looks like the new implementation is about 2x faster, mostly
because:

 - no allocations of `ReadResult()`
 - using `utf8Slice()` directly.

I think @rf might be working on porting this fixes back
to `bufrw` itself; this will be great.

r: @jcorbin @rf

### CODE REVIEW

File: v2/call.js [hunk 1]
```diff
@@ -100,19 +122,48 @@ CallRequest.RW.lazy.readService = function lazyReadService(frame) {
...
+        return frame.cache.serviceStr;
+    }
+
+    if (frame.size < CallRequest.RW.lazy.serviceOffset + 1) {
+        return null;
+    }
+    var strLength = frame.buffer.readUInt8(
+        CallRequest.RW.lazy.serviceOffset, false
+    );
+    var end = CallRequest.RW.lazy.serviceOffset + 1 + strLength;
```
Comment (jcorbin) [inline v2/call.js]: use the cache value if set rather than reading it

File: v2/call.js [hunk 2]
```diff
@@ -100,19 +122,48 @@ CallRequest.RW.lazy.readService = function lazyReadService(frame) {
...
-        return res;
+    if (frame.cache.headerStartOffset !== null) {
+        offset = frame.cache.headerStartOffset;
+    } else {
+        // SKIP service~1
+        var res = bufrw.str1.sizerw.readFrom(frame.buffer, offset);
+        if (res.err) {
+            return res;
+        }
+        offset = res.offset + res.value;
```
Comment (jcorbin) [inline v2/call.js]: cache the value for the other path

**Comment by Raynos** [comment]
Addressed review feedback.

File: v2/call.js [hunk 3]
```diff
@@ -100,19 +122,54 @@ CallRequest.RW.lazy.readService = function lazyReadService(frame) {
...
-
-    // SKIP service~1
-    var res = bufrw.str1.sizerw.readFrom(frame.buffer, offset);
-    if (res.err) {
-        return res;
+    if (frame.cache.headerStartOffset !== null) {
+        offset = frame.cache.headerStartOffset;
+    } else {
+        // SKIP service~1
+        var res = bufrw.str1.sizerw.readFrom(frame.buffer, offset);
```
Comment (rf) [inline v2/call.js]: Can we avoid the `res` object alloc here? Why not just a standard `Buffer#readUint8` like above?

Comment (Raynos) [inline v2/call.js]: This is the old path; I'm not updating it in this PR. Just adding offset caching.

**Pull request merged**